lib/http/response.rb

```diff
@@ -96,8 +96,13 @@ def flush
     #   (not an integer, e.g. empty string or string with non-digits).
     # @return [Integer] otherwise
     def content_length
+      # http://greenbytes.de/tech/webdav/rfc7230.html#rfc.section.3.3.3
+      # Clause 3: "If a message is received with both a Transfer-Encoding
+      # and a Content-Length header field, the Transfer-Encoding overrides the Content-Length.        
+      return nil if @headers.include?(Headers::TRANSFER_ENCODING) ||
+                    !@headers.include?(Headers::CONTENT_LENGTH)
```

Comment:
I would split those in two separate statements for sake of simplicity (took me a while to understand that this is correct statement -- was pretty sure there's a mistake) as it was.

``` ruby
return nil if @headers.include?(Headers::TRANSFER_ENCODING)

value = @headers[Headers::CONTENT_LENGTH]
return nil unless value

# ...
```

Calling `#include?` and then `#[]` will involve extra call in case of "happy path". Although `#include?` is more efficient than `#[]`, when you end up having those two - it worth to keep `#[]` only (at least in this particular place). Just to explain:

``` ruby
def Headers#include?(name)
  @headers_array.any? { |key, val| key == name }
end

def Headers#[](name)
  values = @headers_array.each_with_object([]) { |(k, v), a| a << v if k == name }
  return nil if values.empty?
  return values.first if 1 == values.count
  values
end
```